lib/std/ascii.zig

```diff
@@ -182,20 +182,14 @@ pub const isASCII = isAscii;
 
 /// Uppercases the character and returns it as-is if already uppercase or not a letter.
 pub fn toUpper(c: u8) u8 {
-    if (isLower(c)) {
-        return c & 0b11011111;
-    } else {
-        return c;
-    }
+    const mask = @as(u8, @intFromBool(isLower(c))) << 5;
+    return c ^ mask;
```

Comment:
Shouldn't this be `&`, rather than `^` (xor)? As is it would toggle the case, which isn't what the function is supposed to do (both by convention and according to doc comment)

Reply:
xor looks correct -- you're right that it toggles the case, but only when isLower(c) is true